### ISSUE TITLE
IPv6 support

### ISSUE THREAD

**Issue by mythic-beasties** [irrtoolset/irrtoolset#42]
On an IPv6 only machine with NAT64,

# rtconfig -h whois.ripe.net -protocol ripe  -cisco_use_prefix_lists < scylla.rt 
Connection to whois.ripe.net:43 failed!

# telnet whois.ripe.net 43
Trying 2001:67c:2e8:22::c100:687...
Connected to whois.ripe.net.
Escape character is '^]'.
% This is the RIPE Database query service.
% The objects are in RPSL format.
%
% The RIPE Database is subject to Terms and Conditions.
% See http://www.ripe.net/db/support/db-terms-conditions.pdf

**Comment by nickhilliard** [comment]
tl;dr: this is unlikely to be fixed unless you provide patches and fix some of the underlying structural problems in irrtoolset.

longer answer: without intending to, you've just stood on a turd.

There are two class libraries in irrtoolset for communicating with whois servers: RAWhoisClient and BirdWhoisClient.  RA handles the Merit IRRd and Bird is the name of the RIPE protocol.  Unfortunately, someone a very long time ago decided it would be ridiculously smart to OO-enable the bsd socket library, so they wrote a framework Socket() class, implemented the usual system-level socket calls in this class library, and then caused BirdWhoisClient to inherit Socket().

The correct fix for this would be to fully disentangle BirdWhoisClient from Socket(), delete the entire Socket class library, then implement the reference code in the BSD getaddrinfo() man page in both the BirdWhoisClient and RAWhoisClient libraries to iterate through all addresses returned by getaddrinfo() when looking up the hostname.  This would be very straightforward in RAWhoisClient, but more troublesome in BirdWhoisClient.

This isn't rocket science or anything, but given the state of irrtoolset development, I doubt it's going to end up high on anyone's priority list.  TBH, unless you have specific requirements which can only be handled by irrtoolset, I'd advise using bgpq3 instead.

Reference BSD code from getaddrinfo() man page:

```
struct addrinfo hints, *res, *res0;
int error;
int s;
const char *cause = NULL;

memset(&hints, 0, sizeof(hints));
hints.ai_family = PF_UNSPEC;
hints.ai_socktype = SOCK_STREAM;
error = getaddrinfo("www.kame.net", "http", &hints, &res0);
if (error) {
        errx(1, "%s", gai_strerror(error));
        /*NOTREACHED*/
}
s = -1;
for (res = res0; res; res = res->ai_next) {
        s = socket(res->ai_family, res->ai_socktype,
            res->ai_protocol);
        if (s < 0) {
                cause = "socket";
                continue;
        }

        if (connect(s, res->ai_addr, res->ai_addrlen) < 0) {
                cause = "connect";
                close(s);
                s = -1;
                continue;
        }

        break;  /* okay we got one */
}
if (s < 0) {
        err(1, "%s", cause);
        /*NOTREACHED*/
}
freeaddrinfo(res0);
```
